docs/Java-SpringBoot-Developer.md

```diff
@@ -0,0 +1,355 @@
+public class Movie  extends  MovieBase{
+
+  @JsonIgnore
+  @Id
+  private String id;
+
+  @PartitionKey
+  private String partitionKey;
+
+  private double rating;
+  private long votes;
+  private long totalScore;
+  private String textSearch;
+  private List<Role> roles;
+
+}
+```
+
+Then query by field name example findByMovieId as below:
+
+[MoviesService.java](https://github.com/jyotsnaravikumar/helium-java/blob/CSE-feedbacks/src/main/java/com/microsoft/azure/helium/app/movie/MoviesService.java)
+
+A good CosmosDB best practice is to compute the partition key from the ID. In our case, we use the integer portion of the Movie or Actor document mod 10. This gives us 10 partitions ("0" - "9") with good distribution. For a deeper discussion on the document modeling decisions, please read this [document](https://github.com/4-co/imdb)
+
+```java
+
+@Service
+public class MoviesService {
+
+    @Autowired
+    MoviesRepository repository;
+
+    public Optional<Movie> getMovie(String movieId) {
+        if (StringUtils.isEmpty(movieId)) {
+            throw new NullPointerException("movieId cannot be empty or null");
+        }
+        //queries by partitionid - partitionkey is the field annotated with @partitionkey
+        List<Movie> movies = repository.findByMovieId(movieId);
+        //queries without partitionkey
+        //repository.findById(movieId);
+        if (movies.isEmpty()) {
+            return Optional.empty();
+        } else {
+            return Optional.of(movies.get(0));
+        }
+    }
+}
+
+```
+
+In order to directly read a document using 1 RU (assuming the document is 1K or less), you need the document's ID and partition key. A good CosmosDB best practice is to compute the partition key from the ID. In our case, we use the integer portion of the Movie or Actor document mod 10. This gives us 10 partitions ("0" - "9") with good distribution. For a deeper discussion on the document modeling decisions, please read this [document](https://github.com/4-co/imdb)
+
+### JAVA-SPRINGBOOT-COSMOSDB-SDK-GAP : support for usage GetRequestCharge​ metrics
+
+GetRequestCharge metrics are not supported in spring-boot-cosmosdb sdk.
+
+### JAVA-SPRINGBOOT-COSMOSDB-SDK-GAP : No native query support with @Query Annotation
+
+ ```java
+ @Repository
+public interface MoviesRepository extends DocumentDbRepository<Movie, String>  {
+    List<Movie> findByMovieId(String movieId);
+    List<Movie> findByTextSearchContaining(String movieName);
+    //Native queries example commented below are not supported by spring-boot-cosmosdb sdk
+    //@Query(“Select * from c where c.year= :1 order by order by c.year ”)
+    //List<Movie> findMoviesByYear(int year);
+
+}
+ ```
+
+#### Solution
+
+- Customers are moving to Cosmos Mongo for easier developer experience with features on Native queries, Cross doc queries, paginations, flexibility support
+
```

Comment:
We shouldn't be recommending the Mongo API on Cosmos. Does it make sense to remove this section?

Reply:
Ok will remove it